apps/backend/src/factory/xlsx/fap.ts

```diff
@@ -74,192 +79,92 @@ const sortByRankOrAverageScore = (data: RowObj[]) => {
-          `Fap with ID '${fapId}'/` +
-            `Call with ID '${callId}/'` +
-            `Instruments with IDs '${instrumentIds}' not found, or the user has insufficient rights`
+    const rows = await Promise.all(
+      records.map(async (proposal) => {
+        const pi = await baseContext.queries.user.getBasic(
+          user,
+          proposal.proposer_id
         );
-      }
-
-      return Promise.all(
-        fapProposalPks.map(({ proposalPk }) =>
-          baseContext.queries.proposal.dataSource.get(proposalPk)
-        )
-      );
-    })
-  );
-
-  const proposalsReviews = await Promise.all(
-    instrumentsProposals.map((proposals) => {
-      return Promise.all(
-        proposals.map((proposal) =>
-          proposal
-            ? baseContext.queries.review.reviewsForProposal(user, {
-                proposalPk: proposal.primaryKey,
-                fapId: fapId,
-              })
-            : null
-        )
-      );
-    })
-  );
 
-  const proposalsTechnicalReviews = await Promise.all(
-    instrumentsProposals.map((proposals) => {
-      return Promise.all(
-        proposals.map((proposal) =>
-          proposal
-            ? baseContext.queries.review.technicalReviewsForProposal(
-                user,
-                proposal.primaryKey
-              )
-            : null
-        )
-      );
-    })
-  );
-
-  const proposalsFapMeetingDecisions = await Promise.all(
-    instrumentsProposals.map((proposals) => {
-      return Promise.all(
-        proposals.map((proposal) =>
-          proposal
-            ? baseContext.queries.fap.getProposalFapMeetingDecisions(user, {
-                proposalPk: proposal.primaryKey,
-                fapId: fapId,
-              })
-            : null
-        )
-      );
-    })
-  );
-
-  const proposalsPrincipalInvestigators = await Promise.all(
-    instrumentsProposals.map((proposals) => {
-      return Promise.all(
-        proposals.map((proposal) =>
-          proposal
-            ? baseContext.queries.user.getBasic(user, proposal.proposerId)
-            : null
-        )
-      );
-    })
-  );
+        const proposalAnswers =
+          await baseContext.queries.questionary.getQuestionarySteps(
+            user,
+            proposal.questionary_id
+          );
 
-  const instrumentProposalsAnswers = await Promise.all(
-    instrumentsProposals.map((proposals) => {
-      return Promise.all(
-        proposals.map((proposal) =>
-          proposal
-            ? baseContext.queries.questionary.getQuestionarySteps(
-                user,
-                proposal.questionaryId
-              )
-            : null
-        )
-      );
-    })
-  );
+        const reviews = await baseContext.queries.review.reviewsForProposal(
+          user,
+          { proposalPk: proposal.proposal_pk, fapId: fapId }
+        );
 
-  const out: FapXLSXData = [];
+        return fapDataRow(
+          proposal.proposal_pk,
+          `${pi?.firstname} ${pi?.lastname}`,
```

Comment:
It will be cleaner if we extract this in it's own variable called something like `piFullName` and just pass it here